[build.webkit.org] [GTK][WPE] Add new buildbots for Ubuntu 24.04

dpino opened the pull request:
#### 4237fe6baabe351e022f35db3deaac8e41117e95
<pre>
[build.webkit.org] [GTK][WPE] Add new buildbots for Ubuntu 24.04
<a href="https://bugs.webkit.org/show_bug.cgi?id=273384">https://bugs.webkit.org/show_bug.cgi?id=273384</a>

Reviewed by Carlos Alberto Lopez Perez.

Now Ubuntu LTS build bots build Ubuntu 24.04 and there&apos;s a specific
entry for Ubuntu 22.04. The Ubuntu 20.04 bots have been removed.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/glib/dependencies/apt:
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/jhbuild/jhbuild-wrapper:
* Tools/jhbuild/jhbuildrc_common.py:
(meson_version):
(init):

Canonical link: <a href="https://commits.webkit.org/278510@main">https://commits.webkit.org/278510@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a8bbcaf108e180fca4b6025984b52d340ff3ba2b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50816 "Failed to checkout and rebase branch from PR 27858") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30113 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3134 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54075 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1507 "Built successfully") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/53119 "Failed to checkout and rebase branch from PR 27858") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36387 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1157 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/41399 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/52915 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/27763 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/43778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/22527 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/25108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/1034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9257 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47089 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/1096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55669 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25918 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/48810 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/50837 "Failed to checkout and rebase branch from PR 27858") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27175 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/43856 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/47887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28043 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7351 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26907 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

clopez commented on Tools/jhbuild/jhbuild-wrapper:
```diff
@@ -24,7 +24,7 @@ import shlex
 import sys
 
-jhbuild_revision = '048920f64d82779eef88158455bfd1d160a3be1a'
+jhbuild_revision = 'master'
```
Please don't set master.

That introduces the risk of random regressions because we can't be sure what is pulled from upstream is the same always.
Is ok to update to the last current master, but use the git hash rather than the master key.

clopez commented on Tools/CISupport/build-webkit-org/config.json:
```diff
@@ -430,9 +430,8 @@
-                    "name": "GTK-Linux-64-bit-Release-Ubuntu-2004-Build", "factory": "NoInstallDependenciesBuildFactory",
+                    "name": "GTK-Linux-64-bit-Release-Ubuntu-2204-Build", "factory": "NoInstallDependenciesBuildFactory",
                     "platform": "gtk", "configuration": "release", "architectures": ["x86_64"],
-                    "additionalArguments": ["--no-experimental-features"],
```
I don't think we should remove this flag from the Debian/Ubuntu bots.
The idea of this stable bots is to test with the default build configuration of the release (the tarballs).
AFAIK experimental features are only enabled by default on developer builds and they tend to require bleeding edge dependencies that are usually not easily available on stable distros.

dpino replied:
I agree; I removed this setting unintentionally.

dpino committed:
[build.webkit.org] [GTK][WPE] Add new buildbots for Ubuntu 24.04
https://bugs.webkit.org/show_bug.cgi?id=273384

Reviewed by Carlos Alberto Lopez Perez.

Now Ubuntu LTS build bots build Ubuntu 24.04 and there's a specific
entry for Ubuntu 22.04. The Ubuntu 20.04 bots have been removed.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/glib/dependencies/apt:
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/jhbuild/jhbuild-wrapper:
* Tools/jhbuild/jhbuildrc_common.py:
(meson_version):
(init):

Canonical link: https://commits.webkit.org/278510@main

clopez commented on Tools/CISupport/build-webkit-org/config.json:
```diff
@@ -442,9 +442,8 @@
-                    "name": "GTK-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004", "factory": "BuildAndGenerateMiniBrowserBundleFactory",
+                    "name": "GTK-Linux-64bit-Release-Packaging-Nightly-Ubuntu2204", "factory": "BuildAndGenerateMiniBrowserBundleFactory",
                     "platform": "gtk", "configuration": "release", "architectures": ["x86_64"],
-                    "additionalArguments": ["--no-bubblewrap-sandbox"],
```
I overlooked this one.

We should disable the sandbox on the packaging bots, otherwise the distributed bundle does not work.
